kore.maude

```diff
@@ -220,18 +227,30 @@ Questions about this definition:
 - do we really need ~> as a K construct?  Can it be a label and thus the way K
   deal with evaluation contexts just a particular methodology and not an
   intrisic part of K ? Same question for =>.
+  Cosmin: Syntacticly at least, I would say "no" for both. From the implementation
+  perspective, this will simplify traversal code even further. 
 - rename .::K and .::KList to . ?  if we get rid of ~> as above, then we may
   only need .::KList.
+  Cosmin: I agree.
 - Should we allow empty productions instead of sort declarations?  A pro
   argument is that we have fewer syntactic constructs.  A cons argument is
   that a sort declaration may be regarded as a different entityr, with
   different kinds of attributes.
+  Cosmin: I don't quite follow.
 - Should we add constant AttributeKeys which have special semantics, like
   "strict", as constants?
+  Cosmin: we may want to discuss the attribute syntax separately.
 - Brackets and disappearing injections are the same thing, arent's they?  A
   production which only has one non-terminal and the production disappears
   from the AST.
+  Cosmin: Yes, I also think they can be seen as the same thing. That being
+          said, I would really like for injections to not dissapear. If something
+          is meant to dissapear, it should be declared as a subsort. And the
+          dissapearing injections (including subsorts) could be done via rules as
+          we discussed before.
```

Comment:
Yes, I'd prefer to keep KORE simpler.  Whatever KORE we end up with after this discussion, is what I plan to call "the K Standard" that all implementations should support.  I can easily see implementations which do not even want to have a "bracket" attribute.

Regarding the subsort vs production, I would have a bold proposal: let us get rid of subsorts all together from KORE, and let implementations decide on their own if they want to implement subsorts.  That is, let us simply require that each production has a klabel, including all the injection productions.  The injections will also need to be strict.

From a theoretical perspective, the order-sorted algebra community is well aware of the fact that subsorts can be simulated with injections, specifically by adding for any sorts S1 < S2 < S3 equations
```
eq injS2inS3(injS1inS2(X:S1)) = injS1inS3(X)
```
and then everything happens modulo such equations, including matching.  The rest is just implementation detail, and I believe that some may not want to implement subsorts in their backend; for example the Coq backend.